admin/assets/javascripts/table-sorting.js

```diff
@@ -1,28 +1,58 @@
 /**
  * Table sorting.
  */
+
 /* global $ */
 $(function () {
   'use strict'
+  if (!window.GOVUK) {
+    window.GOVUK = {}
+  }
+  window.GOVUK.tableSort = {
 
-  /**
-   * Performs sorting for the supplied table
-   * @param {String} tableId
-   */
-  var applySorting = function (tableId) {
-    var getCellValue = function (tr, idx) {
+    getCellValue: function (tr, idx) {
       return tr.children[idx].innerText || tr.children[idx].textContent
-    }
+    },
 
-    var comparer = function (idx, asc) {
+    comparer: function (idx, asc, config) {
       return function (a, b) {
         return (function (v1, v2) {
-          return v1 !== '' && v2 !== '' && !isNaN(v1) && !isNaN(v2) ? v1 - v2 : v1.toString().localeCompare(v2)
-        })(getCellValue(asc ? a : b, idx), getCellValue(asc ? b : a, idx))
+          return v1 !== '' && v2 !== '' && !isNaN(v1) && !isNaN(v2)
+            ? v1 - v2 : window.GOVUK.tableSort.getStringComparisonResult(v1, v2, asc, config)
+        })(window.GOVUK.tableSort.getCellValue(a, idx), window.GOVUK.tableSort.getCellValue(b, idx))
       }
-    }
+    },
+
+    getStringComparisonResult: function (a, b, isAscending, config) {
+      if (this.isNullString(a, config)) {
+        return 1
+      } else if (this.isNullString(b, config)) {
+        return -1
+      } else if (a === b) {
+        return 0
+      } else if (isAscending) {
+        return a.toString().localeCompare(b)
+      } else if (!isAscending) {
+        return b.toString().localeCompare(a)
+      }
+    },
+
+    isNullString: function (v, config) {
+      if (!config.sortNullsLast) {
+        return false
+      }
+      if ((v === undefined || v === null || v === '')) {
+        return true
+      }
+      var hasIgnoredString = config.ignoredStrings.some(function (ignoredString) {
+        return ignoredString === v
+      })
+      if (hasIgnoredString) {
+        return true
+      }
```

Comment:
This is nice, but it seems to me there is a missing `return false` from the end of the function.

Reply:
@jon-shipley done